src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -197,30 +189,39 @@ export function createModelForService(
                 (value: InputParameter) => value.IsApiVersion
             );
             if (apiVersionIndex !== -1) {
-                const apiVersionInOperation = op.Parameters[apiVersionIndex];
-                if (!apiVersionInOperation.DefaultValue?.Value) {
-                    apiVersionInOperation.DefaultValue =
-                        apiVersionParam.DefaultValue;
-                }
-                /**
-                 * replace to the global apiVersion parameter if the apiVersion defined in the operation is the same as the global service apiVersion parameter.
-                 * Three checkpoints:
-                 * the parameter is query parameter,
-                 * it is client parameter
-                 * it does not has default value, or the default value is included in the global service apiVersion.
-                 */
-                if (
-                    apiVersions.has(
-                        apiVersionInOperation.DefaultValue?.Value
-                    ) &&
-                    apiVersionInOperation.Kind ===
-                        InputOperationParameterKind.Client &&
-                    apiVersionInOperation.Location === apiVersionParam.Location
-                ) {
-                    op.Parameters[apiVersionIndex] = apiVersionParam;
+                if (apiVersionParam === undefined) {
+                    $lib.reportDiagnostic(program, {
+                        code: "No-APIVersion",
+                        format: { service: service.type.name },
+                        target: NoTarget
+                    });
+                    break;
+                } else {
+                    const apiVersionInOperation =
+                        op.Parameters[apiVersionIndex];
+                    if (!apiVersionInOperation.DefaultValue?.Value) {
+                        apiVersionInOperation.DefaultValue =
+                            apiVersionParam.DefaultValue;
+                    }
+                    /**
+                     * replace to the global apiVersion parameter if the apiVersion defined in the operation is the same as the global service apiVersion parameter.
+                     * Three checkpoints:
+                     * the parameter is query parameter,
```

Comment:
What if api-version is defined in path instead of query parameter?

Reply:
This is how we define api-version as path parameter:
```
@server(
  "{Endpoint}/anomalydetector/{ApiVersion}",
  "",
  {
    Endpoint: string,
    @path
    ApiVersion: APIVersion,
  }
)
```
Then we don't need to define apiversion in operation level.